app/lib/shared/urls.dart

```diff
@@ -143,8 +143,9 @@ String pkgActivityLogUrl(String package) =>
     pkgPageUrl(package, pkgPageTab: PkgPageTab.activityLog);
 
 String pkgArchiveDownloadUrl(String package, String version, {Uri? baseUri}) {
-  final path =
-      '/packages/${Uri.encodeComponent(package)}/versions/${Uri.encodeComponent(version)}.tar.gz';
+  final p = Uri.encodeComponent(package);
+  final v = Uri.encodeComponent(version);
+  final path = '/api/archives/$p-$v.tar.gz';
```

Comment:
We should probably use `Uri.encodeComponent` here, just to be safe.

Reply:
Nevermind, I didn't read two lines up.

Reply:
We do that on the line above, right? :rofl: